Utilities/XMLDeserializer.cs

```diff
@@ -0,0 +1,115 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Xml;
+using System.Xml.Linq;
+using System.Xml.Schema;
+using System.Xml.Serialization;
+using QCUtilities.Entities;
+using QCUtilities.Interfaces;
+
+namespace QCUtilities
+{
+    public class XMLDeserializer
+    {
+        private readonly IXMLFileValidator fileValidator;
+
+        public XMLDeserializer(IXMLFileValidator fMan)
+        {
+            fileValidator = fMan;
+        }
+
+        public List<Post> DeserializeXML(string fileName)
+        {
+            ValidateXML(fileName);
+
+            var ser = new XmlSerializer(typeof(PostCollection));
+            var ps = new PostCollection();
+            using (Stream reader = new FileStream(fileName, FileMode.Open))
+            {
+                ps = (PostCollection)ser.Deserialize(reader);
+            }
+
+            return ps.ToList();
+
+        }
+
+       
+
+        private void ValidateXML(string fileName)
+        {
+            if (!fileValidator.IsValidFileName(fileName))
+            {
+                throw new ArgumentOutOfRangeException($"Invalid fileName. Make sure it is an .xml file. Filename {fileName}");
+            }
+            if (!fileValidator.FileExists(fileName))
+            {
+                throw new FileNotFoundException($"Invalid fileName. Make sure it is an .xml file");
+            }
+            if (!fileValidator.IsFileValidXML(fileName))
+            {
+                throw new InvalidDataException($"File is not a valid xml!");
+            }
+        }
+        
+    }
+
+    public class FileValidator : IXMLFileValidator
```

Comment:
I admit this whole thing feels like overdesign to me; we're only going to be loading a file in one place, we can just throw up and exit out if we need to. This is the kind of thing where "catch generic exception, exit if there's an error" is perfectly legit. I'm also not sure why we need an interface for it - it's a stub for testing's sake? But at that point we've spent like three hundred lines of code implementing an interface and testing it, where we could just load two files with hardcoded names and test a schema.

The stuff you've built here is valuable for API barriers where we're trying to provide diagnostic information to someone who doesn't care about digging through sourcecode, but this isn't an API barrier, the only people who are likely to use this for a while are going to be us two, and it's all slated for eventual demolition anyway when it's more fleshed out and featureful.

I don't think it's necessarily wrong to have this in place, and now that it's here we may as well keep it, but remember that your time is valuable! Spend it on the things that give the biggest bang for buck.